### PR TITLE
apply Modal, insert star Point, create related actions and reducers,apply passport setting and apply passport authentication in progress etc..

### PR DESCRIPTION
**Pull request by Unini**
apply Modal, insert star Point, create related actions and reducers,apply passport setting and apply passport authentication in progress etc..

### CODE REVIEW

File: bookflex-client/src/actions/index.js [hunk 1]
```diff
@@ -63,18 +63,66 @@ export const onClickTab = (key) => {
...
+        dispatch({
+          type: 'BESTSELLER_POSTS',
+          payload: json
+        })
       })
-   }
+  };
 }
 
+export function insertStarPoint(bookId, starPoint) {
```
Comment (breakstorm) [inline bookflex-client/src/actions/index.js:78]: part of rate star

File: bookflex-client/src/actions/index.js [hunk 2]
```diff
@@ -63,18 +63,66 @@ export const onClickTab = (key) => {
...
+      method: 'POST',
+      headers: new Headers({
+        'Content-Type': 'application/json'
+      }),
+      body: JSON.stringify({bookId: bookId, starPoint: starPoint})
+    })
+  };
+}
+
+export function registerUser(email, password) {
```
Comment (breakstorm) [inline bookflex-client/src/actions/index.js:95]: part of registerUser

File: bookflex-client/src/actions/index.js [hunk 3]
```diff
@@ -63,18 +63,66 @@ export const onClickTab = (key) => {
...
+      }),
+      body: JSON.stringify({email: email, password: password})
+    }).then(response => response.json())
+      .then(json => {
+        console.log(json);
+      })
+  };
+}
 
+export function confirmUser(email, password) {
```
Comment (breakstorm) [inline bookflex-client/src/actions/index.js:110]: part of LoginUser

File: bookflex-client/src/actions/index.js [hunk 4]
```diff
@@ -63,18 +63,66 @@ export const onClickTab = (key) => {
...
 
+    fetch(`http://localhost:3001/confirm/user?user=${email}&password=${password}`)
+      .then(response => response.json())
+      .then(json => {
+        console.log("my json", json);
+      })
+  };
+}
 
+export function logoutUser(req, res) {
```
Comment (breakstorm) [inline bookflex-client/src/actions/index.js:122]: part of LogoutUser

File: bookflex-client/src/components/Modal.js [hunk 5]
```diff
@@ -5,9 +5,13 @@ import React, {Component} from 'react';
 
 export default class Modal extends Component {
```
Comment (breakstorm) [inline bookflex-client/src/components/Modal.js:6]: part of Modal component.

File: bookflex-client/src/components/Rating.js [hunk 6]
```diff
@@ -7,10 +7,14 @@ import Stars from './Stars';
 
 export default class Rating extends Component {
```
Comment (breakstorm) [inline bookflex-client/src/components/Rating.js:8]: part of rateComponent

File: bookflex-client/src/components/Stars.js [hunk 7]
```diff
@@ -2,24 +2,43 @@
  * Created by Joy on 2017. 5. 10..
  */
 import React, {Component} from 'react';
```
Comment (breakstorm) [inline bookflex-client/src/components/Stars.js:4]: part of StartComponent

**Review by wakoong — APPROVED** [review]

**Review by breakstorm — APPROVED** [review]
check new feature.

**Pull request merged**